packages/haul-preset-0.59/src/defaultConfig.ts

```diff
@@ -197,12 +199,13 @@ export default function getDefaultConfig(
     optimization: {
       minimize: !!minify,
       minimizer: [
-        // new UglifyJsPlugin({
-        //   test: /\.(js|(js)?bundle)($|\?)/i,
-        //   cache: true,
-        //   parallel: true,
-        //   sourceMap: true,
-        // }),
+        new TerserWebpackPlugin({
+          test: /\.(js|(js)?bundle)($|\?)/i,
+          cache: true,
+          // Set upper limit on CPU cores, to prevent Out of Memory exception on CIs.
+          parallel: Math.max(os.cpus().length, 4) - 1,
```

Comment:
It's a bad idea to limit everybody because of the CI specifics. Use the `is-ci` module do determine if you're in the CI and then limit